DecodeSample/DataToWriter.cs

```diff
@@ -134,15 +134,7 @@ public void WritePerfData(string perfDataFilePath)
                             // formats. TraceFS fields are always scalars or arrays of fixed-size elements, so
                             // the following will work to get the data as a JSON value.
                             this.scratch.Clear();
-                            if (fieldValue.Type.IsArrayOrElement)
-                            {
-                                fieldValue.AppendJsonSimpleArrayTo(this.scratch);
-                            }
-                            else
-                            {
-                                fieldValue.AppendJsonScalarTo(this.scratch);
-                            }
-
+                            fieldValue.AppendJsonTo(this.scratch);
```

Comment:
Note: Old code was using `IsArrayOrElement` as a partial substitute for `!IsScalar` (worked in this case since TraceFS fields cannot be element, only scalar or array). Now the fieldValue knows whether it is a scalar, so we can just call AppendJsonTo and it will do the right thing.